src/core/energy_storage/electric_vehicle.jl

```diff
@@ -0,0 +1,342 @@
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+# Redistributions of source code must retain the above copyright notice, this list
+# of conditions and the following disclaimer.
+#
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or other
+# materials provided with the distribution.
+#
+# Neither the name of the copyright holder nor the names of its contributors may be
+# used to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+# *********************************************************************************
+"""
+    ElectricVehicle
+
+Inputs used when:
+```julia 
+haskey(d["ElectricVehicle"])
+```
+Defined by these parameters which are contained in the EV dictionary, unique to EV vs (stationary) ElectricStorage
+```julia
+Base.@kwdef mutable struct ElectricVehicle
+    energy_capacity_kwh::Float64 = NaN
+    max_c_rate::Float64 = 1.0
+    ev_on_site_start_end::Array{Int64, 1} = [0,0]  # This should **maybe** be an array of arrays (meant to be array of tuples, could convert)
+    ev_on_site_series::Array{Int64, 1} = []
+    soc_used_off_site::Float64 = 0.0
+    leaving_next_time_step_soc_min::Array{Float64, 1} = []
+    back_on_site_time_step_soc_drained::Array{Float64, 1} = []
+end
+```
+
+# TODO
+- Align schedule parameters with year, and day of week = 1 is Sunday or Monday?
+- Weeks or days where EV is never on-site
+- Add soc_used_off_site options to be Scalar, 2x (weekday/weekend), or 7x (each day of week) Values of 0.0 to (1.0 - min_soc)
+
+
+"""
+Base.@kwdef mutable struct ElectricVehicle
+    energy_capacity_kwh::Float64 = NaN
+    max_c_rate::Float64 = 1.0
+    ev_on_site_start_end::Array{Array{Int64, 1}, 1} = [[0,0],[0,0],[0,0],[0,0]]  # This should **maybe** be an array of arrays (meant to be array of tuples, could convert)
+    ev_on_site_series::Union{Array{Int64, 1}, Nothing} = nothing
+    soc_used_off_site::Array{Array{Float64, 1}, 1} = [[0.2,0.8],[0.2,0.8],[0.2,0.8],[0.2,0.8]]
```

Comment:
@rathod-b what do the two values in the inner arrays mean? I was thinking there's just one "SOC used" value for each quarter. The ev_on_site_start_end have two values because you need the start and end timestep.

Reply:
The pair of values in each array can represent the typical incoming EV SOC and desired outbound EV SOC for each quarter. I was thinking that the outbound SOC can be different based on quarters, allowing us to model higher desired SOCs in the summer and winter due to HVAC use and comparatively lower desired SOCs in Fall/Spring. Hopefully this makes sense. Maybe I should rename this fieldname to `ev_arrive_depart_soc`?

Reply:
Ok, I thought that might be the case. We currently we also have `leaving_next_time_step_soc_min` and `back_on_site_time_step_soc_drained`  which are placeholders for custom hourly/15-minute data that would be a separate way to enter this data (along with `ev_on_site_series`, but that might be redundant if we have the first two series data). I think you already know this, so no need to respond.